### PR TITLE
Fix clang template warnings

### PR DESCRIPTION
**Pull request by fmatthew5876**
Fix: #221

Re-implementation of #290, this time with autogenerated declarations.

Unfortunately, this is still broken with MSVC 2019, so I put ifdefs around it.

### CODE REVIEW

**Comment by carstene1ns** [comment]
(You could have reused/reopened the old pull request instead)

File: generator/templates/fwd_header.tmpl [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// MSVC incorrectly treats these declarations as definitions and fails.
```
Comment (carstene1ns) [inline generator/templates/fwd_header.tmpl]: Should add:
```jinja2
{% include "copyright.tmpl" %}

```

Comment (fmatthew5876) [reply generator/templates/fwd_header.tmpl]: Fixed

File: src/generated/fwd_struct_impl.h [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+ * This file is part of liblcf. Copyright (c) 2019 liblcf authors.
+ * https://github.com/EasyRPG/liblcf - https://easyrpg.org
+ *
+ * liblcf is Free/Libre Open Source Software, released under the MIT License.
+ * For the full copyright and license information, please view the COPYING
+ * file that was distributed with this source code.
+ */
+
+// MSVC incorrectly treats these declarations as definitions and fails.
+#ifndef _MSC_VER
```
Comment (Ghabry) [inline src/generated/fwd_struct_impl.h]: still broken in VS2019 16.4?

Comment (fmatthew5876) [reply src/generated/fwd_struct_impl.h]: Still broken in 16.4 yes

**Comment by fmatthew5876** [comment]
jenkins test this please

**Pull request merged**